lib/Twig/Cache/Filesystem.php

```diff
@@ -29,11 +29,11 @@ public function __construct($directory)
     /**
      * {@inheritdoc}
      */
-    public function generateKey($className, $prefix)
+    public function generateKey($name, $className)
     {
-        $class = substr($className, strlen($prefix));
+        $len = strlen($className);
 
-        return $this->directory.'/'.$class[0].'/'.$class[1].'/'.$class.'.php';
+        return $this->directory.'/'.$className[$len - 1].'/'.$className[$len - 2].'/'.$className.'.php';
```

Comment:
An architectural better solution would be to hash the class name again and then use two chars from the hash. This would also solve the problem when `strlen($className) < 2`.